pkg/kms/arieskms.go

```diff
@@ -102,16 +119,29 @@ func NewAriesKeyManager(cfg *Config) (*KeyManager, error) {
 
 		awsSvc := awssvc.New(awsSession, nil, "")
 
+		awsSvcKMS := &awsKMSSvc{Service: *awsSvc, metrics: metrics}
+
 		return &KeyManager{
 			kmsType:    cfg.KMSType,
-			keyManager: awsSvc,
+			KeyManager: awsSvcKMS,
 			crypto:     awsSvc,
 		}, nil
 	}
 
 	return nil, fmt.Errorf("unsupported kms type: %s", cfg.KMSType)
 }
 
+type awsKMSSvc struct {
+	awssvc.Service
+
+	metrics metricsProvider
+}
+
+// Create function to support AFGO's KeyManager api.
```

Comment:
why we add create. The task about adding metrics provider

Reply:
this should be removed

Reply:
actually, this is to be able to pass the metrics object to the aws kms service so that it can pass it to the KMSSigner (see line 193 further down in this file). This signer includes the metrics gathering in Sign().

Reply:
will need to refactor this code snippet a bit.. to make it more consistent and tidy.

Reply:
updated, and removed this struct as it's unnecessary.

Reply:
removed